plugins/packages/couchdb/lib/index.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  QueryError,
+  QueryResult,
+  QueryService,
+  ConnectionTestResult,
+} from "@tooljet-plugins/common";
+import { SourceOptions, QueryOptions } from "./types";
+import got from "got";
+// const NodeCouchDb = require('node-couchdb');
+
+export default class Couchdb implements QueryService {
+  async run(
+    sourceOptions: SourceOptions,
+    queryOptions: QueryOptions
+  ): Promise<QueryResult> {
+    let result = {};
+    let response = null;
+    const {
+      operation,
+      record_id,
+      limit,
+      view_url,
+      start_key,
+      end_key,
+      skip,
+      descending,
+    } = queryOptions;
+    const { username, password, port, host, database, protocol } =
+      sourceOptions;
+    const revision_id = queryOptions.rev_id;
+    const authHeader = () => {
+      const combined = `${username}:${password}`;
+      const key = Buffer.from(combined).toString("base64");
+      return { Authorization: `Basic ${key}` };
+    };
+    try {
+      switch (operation) {
+        case "list_records": {
+          response = await got(
+            `${protocol}://${host}:${port}/${database}/_all_docs`,
+            {
+              method: "get",
+              headers: authHeader(),
+              searchParams: {
+                ...(limit?.length > 0 && { limit }),
+                ...(skip?.length > 0 && { skip }),
+                ...(descending && { descending }),
+              },
+            }
+          );
+          result = JSON.parse(response.body);
+          break;
+        }
+        case "retrieve_record": {
+          response = await got(
+            `${protocol}://${host}:${port}/${database}/${record_id}`,
+            {
+              headers: authHeader(),
+              method: "get",
+            }
+          );
+
+          result = JSON.parse(response.body);
+          break;
+        }
+
+        case "create_record": {
+          response = await got(`${protocol}://${host}:${port}/${database}`, {
+            method: "post",
+            headers: authHeader(),
+            json: {
+              records: JSON.parse(queryOptions.body),
```

Comment:
will be good if we use [JSON5](https://json5.org/) to parse user entered json string